build-system/tasks/sweep-experiments/test/test.js

```diff
@@ -0,0 +1,511 @@
+
+  t.is(Object.keys(exclude).length, 1);
+  t.true('swg-special-excluded' in exclude);
+  t.like(exclude['swg-special-excluded'], {percentage: 1});
+
+  t.is(Object.keys(include).length, 3);
+  t.like(include['matching-off'], {percentage: 0});
+  t.like(include['matching-on'], {percentage: 1});
+  t.like(include['introduced-later'], {percentage: 1});
+
+  for (const {previousHistory} of [
+    exclude['swg-special-excluded'],
+    include['matching-off'],
+    include['matching-on'],
+  ]) {
+    t.is(previousHistory.length, 1);
+    const {hash, authorDate, subject} = previousHistory[0];
+    t.truthy(hash);
+    t.truthy(authorDate);
+    t.is(subject, 'Introduce config');
+  }
+
+  t.is(include['introduced-later'].previousHistory.length, 1);
+  const {hash, authorDate, subject} = include[
+    'introduced-later'
+  ].previousHistory[0];
+  t.truthy(hash);
+  t.truthy(authorDate);
+  t.is(subject, 'Add introduced-later');
+});
+
+testTempDir.only('sweep-experiments', async (t, dir) => {
+  t.timeout(999999999); // see below
+
+  gitInit(dir);
+
+  const prodConfig = {
+    'allow-doc-opt-in': ['removed'],
+    'allow-url-opt-in': ['removed'],
+    'removed': 1,
+    'not-removed': 0,
+  };
+
+  const canaryConfig = {
+    'allow-doc-opt-in': ['removed', 'kept'],
+    'allow-url-opt-in': ['removed'],
+    'removed': 1,
+    'not-removed': 1,
+    'also-not-removed': 0,
+  };
+
+  await writeDir(dir, {
+    'build-system/global-configs/prod-config.json': JSON.stringify({}, null, 2),
+    'build-system/global-configs/canary-config.json': JSON.stringify(
+      {},
+      null,
+      2
+    ),
+  });
+  exec('git add .', {cwd: dir});
+  exec('git commit --date=1998-12-31 -m "empty config"', {cwd: dir});
+
+  await writeDir(dir, {
+    'build-system/global-configs/prod-config.json': JSON.stringify(
+      prodConfig,
+      null,
+      2
+    ),
+    'build-system/global-configs/canary-config.json': JSON.stringify(
+      canaryConfig,
+      null,
+      2
+    ),
+    'extensions/uses-is-experiment-on.js': `
+      import {isExperimentOn} from 'wherever';
+      isExperimentOn(window, 'removed');
+      isExperimentOn(window, 'not-removed');
+    `,
+    'test/uses-toggle-experiment.js': `
+      import {toggleExperiment} from 'wherever';
+      toggleExperiment(window, 'removed', true);
+    `,
+  });
+
+  exec('git add .', {cwd: dir});
+  exec('git commit --date=1999-12-31 -m "code written long ago"', {cwd: dir});
+
+  // const whenDoesThisBreak = new Date();
+  // whenDoesThisBreak.setSeconds(whenDoesThisBreak.getSeconds() + 1);
+
+  // await new Promise((resolve) => {
+  //   setTimeout(resolve, 3000);
+  // });
+
+  t.is(
+    getStdout(
+      `git log --until=2021-03-01T08:00:00.812Z -S \'"removed": 1,\'  --format="%h %aI %s" ${dir}/build-system/global-configs/prod-config.json`,
+      // `git log -S \'"removed": 1,\'  --format="%h %aI %s" ${dir}/build-system/global-configs/prod-config.json`,
+      {cwd: dir}
+    ),
+    'foo'
+  );
```

Comment:
- This `git log` keeps returning `''`. 
- `collectWork()` is empty because `days_ago` affects `git log`
- this might be related to setting a `git commit --date` in the past

Reply:
but this returns the commit! and it's in the constant past date! but the command is wrong!
```js
`git log --until=${new Date().toISOString()} -S \'"removed": 1,\'  --format="%h %aI %s" ${dir}/build-system/global-configs/prod-config.json`
```
```
42564c2 1999-12-31T16:39:32-08:00 ...
```